app/models/institutions_archive.rb

```diff
@@ -0,0 +1,56 @@
+# frozen_string_literal: true
+
+require_dependency 'institution'
+
+class InstitutionsArchive < Institution
+  self.table_name = 'institutions_archives'
+
+  # class methods
+  # rubocop:disable Metrics/MethodLength
+  def self.archive_previous_versions(user)
+    production_version = Version.current_production.number
+    previous_version = Version.previous_production.number
+
+    begin
+      ActiveRecord::Base.transaction do
+        create_archives(previous_version, production_version)
+        Institution.where('version >= ? and version < ?', previous_version, production_version).delete_all
+      end
+    rescue ActiveRecord::StatementInvalid => exception
+      notice = 'There was an error occurring at the database level'
```

Comment:
Extract the body of this `rescue`and the one below into a method (since they are identical other than the `notice` value) and call it with the `notice`.
e.g. `process_exception('There was an error occurring at the database level')`

Reply:
Made this change